.github/workflows/ci-backwards-compatibility.yml

```diff
@@ -0,0 +1,83 @@
+# This is a basic workflow to help you get started with Actions
+
+name: "Backwards-Compatibility"
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch and release tags
+  push:
+    branches: [ "main", "master", "devel", "releases/v*" ]
+    tags:
+      - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+  pull_request:
+    branches: [ "main", "master", "devel", "releases/v*" ]
+  merge_group:
+    branches: [ "main", "master", "devel", "releases/v*" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```

Comment:
I would like to avoid keeping it as a PR forever, but I'm worried that even non-required run doesn't buy us anything, will cost money to run, will still cause PRs to wait for it to finish slowing us down and will make the :heavy_check_mark: to turn into :negative_squared_cross_mark: in the UI.

That's why I was thinking about making it available as `just ...` command for now, and maybe we could make it a workflow that one can dispatch manually, and just not run it on PRs at all for now.

But I guess we can land first, see how it goes, and tweak later.

Reply:
re: costs

Some back-of-the-envelope math

- 70 commits to `fedimint/fedimint` in the last month
  - https://github.com/fedimint/fedimint/pulse/monthly
  - Assuming 1 backwards-compatibility run per commit
- Assuming cost is accurately reflected in buildjet's [pricing](https://buildjet.com/for-github-actions/docs/about/pricing), we pay $0.016 / min
- Backwards-compatibility workflow takes ~30 minutes

`0.016 $/min * 30 min * 70 commits/month => ~$34 / month`

Ballpark, $34 / month seems reasonable.

Reply:
Runtime cost isn't our biggest concern imo, but number of blocked runners are (and upping our runner count would be $$$). I'm ok with turning it on and seeing how it goes.